Projects/Contracts/Sources/View/ContractTable.swift

```diff
@@ -9,82 +9,72 @@ import hCoreUI
 import hGraphQL
 
 struct ContractTable {
-    let filter: ContractFilter
     @PresentableStore var store: ContractStore
+    
+    @State
+    var contracts = [Contract]()
+    
+    private func updateContracts(for state: ContractState) {
+        let contracts = store.state.contracts + store.state.contractBundles.flatMap { $0.contracts }
 
-    func getContractsToShow(for state: ContractState, filter: ContractFilter) -> [Contract] {
-        switch filter {
-        case .active:
-            return state
-                .contractBundles
-                .flatMap { $0.contracts }
-        case .terminated:
-            return state.contracts.filter { contract in
-                contract.currentAgreement?.status == .terminated
-            }
-        case .none: return []
+        if contracts.isEmpty {
+            self.contracts = store.state.terminatedContracs
+        } else {
+            self.contracts = contracts
         }
     }
 }
 
 extension ContractTable: View {
     var body: some View {
         ContractBundleLoadingIndicator()
-
+        
         hSection {
-            PresentableStoreLens(
-                ContractStore.self,
-                getter: { state in
-                    getContractsToShow(for: state, filter: filter.nonemptyFilter(state: state))
-                }
-            ) { contracts in
-                ForEach(contracts, id: \.id) { contract in
-                    ContractRow(id: contract.id)
-                        .fixedSize(horizontal: false, vertical: true)
-                        .padding(.top, 15)
-                        .transition(.slide)
-                }
+            ForEach(contracts, id: \.id) { contract in
```

Comment:
Does removing the `PresentableStoreLens` not have any affect when polling for contracts?

Reply:
Not that I am aware of. The polling should update the `state` which is received in the `onReceive` block, which then updates the local `state` value.